### PR TITLE
[TACHYON-1485]split TfsShellTest into files

### PR DESCRIPTION
**Pull request by summerDG**
https://tachyon.atlassian.net/browse/TACHYON-1485
This commit splits TfsShellTest to files so that it's more readable and maintainable.

### CODE REVIEW

**Comment by gpang** [comment]
ok to test

File: integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+import tachyon.shell.command.CommandUtils;
+import tachyon.thrift.FileInfo;
+import tachyon.util.FormatUtils;
+import tachyon.util.io.BufferUtils;
+import tachyon.util.io.PathUtils;
+
+/**
+ * The base class for all the TfsShellTest classes
+ */
+public class AbstractTfsShellTest {
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java]: Could you make this class abstract?

File: integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import tachyon.security.LoginUser;
+import tachyon.security.authentication.AuthType;
+import tachyon.shell.command.CommandUtils;
+import tachyon.thrift.FileInfo;
+import tachyon.util.FormatUtils;
+import tachyon.util.io.BufferUtils;
+import tachyon.util.io.PathUtils;
+
+/**
+ * The base class for all the TfsShellTest classes
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java]: Please add period to all the sentences.

File: integration-tests/src/test/java/tachyon/shell/command/CatDirectoryTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import tachyon.shell.AbstractTfsShellTest;
+
+/**
+ * Test used to cat Directory
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/CatDirectoryTest.java]: Please add period for this sentence and others in this PR.

File: integration-tests/src/test/java/tachyon/shell/command/CreateCacheInsertInUfsThenloadMetadataTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import tachyon.client.TachyonFSTestUtils;
+import tachyon.client.TachyonStorageType;
+import tachyon.client.UnderStorageType;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.TachyonException;
+import tachyon.shell.AbstractTfsShellTest;
+import tachyon.util.UnderFileSystemUtils;
+import tachyon.util.io.PathUtils;
+
+public class CreateCacheInsertInUfsThenloadMetadataTest extends AbstractTfsShellTest {
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/CreateCacheInsertInUfsThenloadMetadataTest.java]: Please add javadoc

File: integration-tests/src/test/java/tachyon/shell/command/FreeTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import tachyon.client.TachyonStorageType;
+import tachyon.client.UnderStorageType;
+import tachyon.client.file.TachyonFile;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.TachyonException;
+import tachyon.shell.AbstractTfsShellTest;
+import tachyon.util.CommonUtils;
+
+/**
+ * Created by dell on 2016/1/15.
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/FreeTest.java]: Please fix the javadoc

File: integration-tests/src/test/java/tachyon/shell/command/LoadMetadataErrorMessageTest.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+package tachyon.shell.command;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import tachyon.exception.ExceptionMessage;
+import tachyon.shell.AbstractTfsShellTest;
+
+/**
+ * Test for loadMetadata command with a error message
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/LoadMetadataErrorMessageTest.java]: typo: a error -> an error

File: integration-tests/src/test/java/tachyon/shell/command/LsThenloadMetadataTest.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import tachyon.exception.ExceptionMessage;
+import tachyon.exception.TachyonException;
+import tachyon.shell.AbstractTfsShellTest;
+import tachyon.util.UnderFileSystemUtils;
+import tachyon.util.io.PathUtils;
+
+/**
+ * Test for ls with a wildcard argument
+ */
+public class LsThenloadMetadataTest extends AbstractTfsShellTest {
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/LsThenloadMetadataTest.java]: I think you group this and `LsWildcardTest` into `LsTest`. Basically each command has its own related tests.

File: integration-tests/src/test/java/tachyon/shell/command/MkdirExistingTest.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import tachyon.shell.AbstractTfsShellTest;
+
+/**
+ * Test for mkdir command with a existing directory
+ */
+public class MkdirExistingTest extends AbstractTfsShellTest {
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/MkdirExistingTest.java]: Similarly, all the mkdir tests can be grouped into one file.

**Comment by yupeng9** [comment]
@summerDG thanks a lot for organizing the tests! I left some comments, and the main one is that I suggest you group tests per command. Also please `@` me after you have addressed the comments.

File: integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java (hunk 1 above)
Comment (summerDG) [inline integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java]: yes,I make TfsShellTest abstract

File: integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java (hunk 2 above)
Comment (summerDG) [inline integration-tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java]: ok

File: integration-tests/src/test/java/tachyon/shell/command/FreeTest.java (hunk 5 above)
Comment (summerDG) [inline integration-tests/src/test/java/tachyon/shell/command/FreeTest.java]: ok

File: integration-tests/src/test/java/tachyon/shell/command/MkdirExistingTest.java (hunk 8 above)
Comment (summerDG) [inline integration-tests/src/test/java/tachyon/shell/command/MkdirExistingTest.java]: @yupeng9 , it is suitable for many command to group tests, but there are many tests contain two or more commands, so could you give me some advice?

Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/shell/command/MkdirExistingTest.java]: If the test name suggests one of the commands is the main one for test, then you add it to the corresponding group. For those few ones consisting of multiple commands to serve some complex operation, then perhaps you can add them to a `complexCommandTest`. What do you think?

Comment (summerDG) [inline integration-tests/src/test/java/tachyon/shell/command/MkdirExistingTest.java]: It is a good idea.

File: integration-tests/src/test/java/tachyon/shell/command/CopyFromLocalOverwriteTest.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import tachyon.TachyonURI;
+import tachyon.client.file.TachyonFile;
+import tachyon.shell.AbstractTfsShellTest;
+import tachyon.util.io.BufferUtils;
+
+/**
+ * Created by dell on 2016/1/15.
```
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/shell/command/CopyFromLocalOverwriteTest.java]: update javadoc

**Comment by aaudiber** [comment]
Thanks for the PR! I agree that tests for the same command should be grouped together in the same test file

**Comment by yupeng9** [comment]
@summerDG  Thanks for addressing the comments. Could you also address the comments in `AbstractTfsShellTest`?

**Comment by summerDG** [comment]
@yupeng9 , if I make TfsShellTest abstract, it will not run cause abstract class cannot run any method by itself. So I want to change it to TfsShellTest, not abstract.

**Comment by yupeng9** [comment]
Jenkins, test this please

**Comment by yupeng9** [comment]
@summerDG  It shouldn't have such problem, because no method in that class is not annotated with `@Test`

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**